src/detail_print.js

```diff
@@ -0,0 +1,113 @@
+console.debug('Injected detail print script');
+
+import { sendBackgroundScriptAMessage, MESSAGE_TYPE } from './coms/communication.js';
+
+/**
+ * Wait for an element to be present on the page.
+ * @param {string} selector - The CSS selector to wait for
+ * @return {Element} The element being waited for
+ */
+async function waitForElement(selector) {
+  let element = document.querySelector(selector);
+  while (!element) {
+    element = document.querySelector(selector);
+    await new Promise((r) => setTimeout(r, 200));
+  }
+  return element;
+}
+
+/**
+ * Returns a value indicating whether the payment is complete.
+ * @return {boolean} Whether payment is completed
+ */
+function isPaymentCompleted() {
+  const payStatus = document.querySelector('tr[ng-if*="report.statusString"] > td').textContent;
+  return payStatus === 'Completed';
+}
+
+/**
+ * Get the details for a reg data table entry.
+ * @param {string} fieldName - The text field name from the reg details table
+ * @return {string} The value of the table
+ */
+function getDetail(fieldName) {
+  const regDetailTable = document.querySelector('div[ng-include*="data.html"] > table');
+  const th = [...regDetailTable.querySelectorAll('th')]
+    .filter((el) => el.textContent.indexOf(fieldName) > -1);
+
+  if (th.length > 0) {
+    return th[0]?.nextElementSibling.textContent ?? '';
+  }
+  return '';
+}
+
+/**
+ * Get an age in years between the birthdate and today.
+ * @param {string} birthdate - The raw birthdate string, in YYYY-mm-dd format.
+ * @param {Date} today - The 'today' value to use. Defaults to new Date().
+ * @return {number} - Age in years between the birthdate and today.
+ */
+function getAge(birthdate, today = new Date()) {
+  // TODO: This method should move to a dedicated Registrant class.
```

Comment:
staaahhhhp

Reply:
If I promise the next PR will include a refactor that moves all 4 (!) of these to a single spot will you let this one through? :)